python/lsst/daf/butler/registry/wildcards.py

```diff
@@ -473,3 +475,89 @@ def __str__(self) -> str:
             terms = list(self.strings)
             terms.extend(str(p) for p in self.patterns)
             return "[{}]".format(", ".join(terms))
+
+
+@dataclasses.dataclass
+class DatasetTypeWildcard:
+    """A validated expression that resolves to one or more dataset types.
+
+    The `from_expression` method should almost always be used to construct
+    instances, as the regular constructor performs no checking of inputs (and
+    that can lead to confusing error messages downstream).
+
+    patterns : `tuple` [ `re.Pattern` ]
```

Comment:
Does this need `Parameters` line?

Reply:
Actually I _think_ I should just remove this parameter doc, because this is a dataclass and `__init__` is machine-generated and the docs are on the attributes.  Or at least that's how I usually handle dataclasses, so I hope that's okay.